libbeat/common/file/file_other.go

```diff
@@ -65,8 +65,13 @@ func ReadOpen(path string) (*os.File, error) {
 
 // IsRemoved checks wheter the file held by f is removed.
 func IsRemoved(f *os.File) bool {
-	_, err := os.Stat(f.Name())
-	return err != nil
+	stat, err := f.Stat()
+	if err != nil {
```

Comment:
The function should rather return a tuple `(bool, error)`. We should not mark a file as remove if the permissions have changed. I'm not sure right now if something else (registrar state/harvester) might be affect if we generously return true here.

Checking the error for a potential removal is the right approach, but instead of `err != nil` we might want to use `os.IsNotExist`.